File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/[C#]-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/C#/WCF_CustomServiceApplication/DayNamerServiceHostFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel.Activation;
using System.ServiceModel;
using Microsoft.SharePoint;

namespace WCF_CustomServiceApplication.Server
{
    //This class enables SharePoint to create multiple service
    //hosts for the service application
    internal sealed class DayNamerServiceHostFactory : ServiceHostFactory
    {
        public override ServiceHostBase CreateServiceHost(string constructorString, Uri[] baseAddresses)
        {
            ServiceHost serviceHost = new ServiceHost(typeof(DayNamerServiceApplication), baseAddresses);

            // configure the service for claims
            serviceHost.Configure(SPServiceAuthenticationMode.Claims);

            return serviceHost;
        }
    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/VB/WCF_CustomServiceApplication/DayNamerServiceHostFactory.vb
Imports System.ServiceModel.Activation
Imports System.ServiceModel
Imports Microsoft.SharePoint


Namespace Server
    'This class enables SharePoint to create multiple service
    'hosts for the service application
    Friend NotInheritable Class DayNamerServiceHostFactory
        Inherits ServiceHostFactory

        Public Overrides Function CreateServiceHost(ByVal constructorString As String, ByVal baseAddresses() As System.Uri) As System.ServiceModel.ServiceHostBase
            Dim serviceHost As ServiceHost = New ServiceHost(GetType(DayNamerServiceApplication), baseAddresses)

            'Configure the service for claims
            serviceHost.Configure(SPServiceAuthenticationMode.Claims)

            Return serviceHost
        End Function

    End Class

End Namespace
